core/src/main/java/cucumber/runtime/arquillian/client/CucumberArchiveProcessor.java

```diff
@@ -355,4 +329,38 @@ private static String featureName(final URL url) {
         // fallback
         return Math.abs(url.hashCode()) + Features.EXTENSION;
     }
+
+    private static boolean isCukeSpaceTest(final TestClass testClass, final ClassLoader loader) {
+        Class<? extends Annotation> runWithType = null;
+        try {
+            runWithType = (Class<? extends Annotation>) loader.loadClass("org.junit.runner.RunWith");
+
+        } catch (final ClassNotFoundException error) {
+            // no-op
+        } catch (final NoClassDefFoundError e) {
+            // no-op
+        }
+
+        if (runWithType == null) {
+            return false;
+        }
+        final Annotation runWith = testClass.getAnnotation(runWithType);
+        if (runWith == null) {
+            return false;
+        }
+
+        try {
+            final Class<?> runner = Class.class.cast(runWithType.getMethod("value").invoke(runWith));
+            if ((!ArquillianCucumber.class.equals(runner) && !CukeSpace.class.equals(runner))) {

```

Comment:
Your proposal sounds good.
Are you ok with  something like: "skipArchiveProcessor" ? 
This could be also useful in case I have test running on both client and server side but I'm using cucumber only  for client one.
And do you agree with having an Helper which expose some methods for the enrichment instead of keeping them inside CucumberArchiveProcessor? 

Thanks
/B